src/Umbraco.Infrastructure/Migrations/MigrationContext.cs

```diff
@@ -22,6 +22,7 @@ public MigrationContext(MigrationPlan plan, IUmbracoDatabase? database, ILogger<
     }
 
     // this is only internally exposed
+    [Obsolete("This will be removed in the V13, and replaced with a RebuildCache flag on the MigrationBase")]
```

Comment:
Looked into this, and I don't think introducing the notification now is a great idea. 

The reasoning is that `Upgrader.Execute` is currently expected to throw an exception if the migration fail. For us to publish the notification we have to alter this behavior and make the `Execute` method never throw, but instead return an extended `ExecutedMigrationPlan` containing a success status and the exception, which would be a breaking change. 

This is required to be able to publish the notification regardless of migration success in `DatabaseBuilder`.

I'll update the other PT also to mention this breaking change since it could be relevant if people create their own migration plans and run them with the upgrader 👍